test/services/database/comment/comment_repository_test.dart

```diff
@@ -0,0 +1,316 @@
+        );
+
+        // Check the post comments are correct
+        final expectedPostComments = [
+          postCommentUser0Post0,
+          postCommentUser1Post0,
+        ];
+        final actualPostComments = await commentRepo.getPostComments(post.id);
+        expect(actualPostComments, unorderedEquals(expectedPostComments));
+
+        // Check the user comments are correct
+        final expectedUserComments = [
+          userCommentUser0Post0,
+          userCommentUser0Post1,
+        ];
+        final actualUserComments = await commentRepo.getUserComments(user.uid);
+        expect(actualUserComments, unorderedEquals(expectedUserComments));
+      });
+
+      test(
+          "should initialize the comment count of a post to 1 when the commentCount field doesn't exist and a post is added",
+          () async {
+        // Remove the comment count field
+        await postDocument
+            .update({PostData.commentCountField: FieldValue.delete()});
+
+        final commentData = postCommentDataGenerator.createMockCommentData();
+
+        await commentRepo.addComment(
+          post.id,
+          commentData,
+        );
+
+        // Check that the comment count was updated correctly
+        final postDoc = await postDocument.get();
+        final actualPost = PostFirestore.fromDb(postDoc);
+
+        expect(actualPost.data.commentCount, equals(1));
+      });
+    });
+
+    /// Utility function to check that the post and user comments are not empty
+    Future<void> checkPostAndUserCommentsNotEmpty() async {
+      final postComments = await commentRepo.getPostComments(post.id);
+      final userComments = await commentRepo.getUserComments(user.uid);
+
+      expect(postComments, isNotEmpty);
+      expect(userComments, isNotEmpty);
+    }
+
+    /// Utility function to check that the post and user comments are empty
+    Future<void> checkPostAndUserCommentsEmpty() async {
+      final postComments = await commentRepo.getPostComments(post.id);
+      final userComments = await commentRepo.getUserComments(user.uid);
+
+      expect(postComments, isEmpty);
+      expect(userComments, isEmpty);
+    }
+
+    Future<(List<UserFirestore>, List<CommentFirestore>)> addCommentsForUsers(
+      int nbUsers,
+    ) async {
+      final users =
+          await FirestoreUserGenerator.addUsers(fakeFirestore, nbUsers);
+      final postComments = <CommentFirestore>[];
+
+      for (user in users) {
+        final commentData =
+            postCommentDataGenerator.createMockCommentData(ownerId: user.uid);
+
+        final commentId = await commentRepo.addComment(post.id, commentData);
+
+        postComments.add(CommentFirestore(id: commentId, data: commentData));
+      }
+
+      // Check that the user comments have been added
+      for (user in users) {
+        final userComments = await commentRepo.getUserComments(user.uid);
+
+        expect(userComments, isNotEmpty);
+      }
+
+      return (users, postComments);
+    }
```

Comment:
This seems close in behaviour with `CommentFirestoreGenerator:addCommentsForUsers`. Is there a way to diminish code repetition?

Reply:
You are right !
I reduce code dupllication in b964bf49e35e46c295a9202a00d36c86f67fc912.